lib/domainsTags.ts

```diff
@@ -101,11 +89,11 @@ export default class DomainTagsClient implements IDomainTagsClient {
     return new DomainTagStatistic(response);
   }
 
-  list(domain: string, query?: DomainTagsQuery): Promise<DomainTagsList> {
-    return this.request.get(urljoin(this.baseRoute, domain, '/tags'), query)
-      .then(
-        (res: APIResponse) => this._parseDomainTagsList(res as DomainTagsResponseData)
-      );
+  async list(domain: string, query?: DomainTagsQuery): Promise<DomainTagsList> {
+    const { updatedQuery, url } = this.updateUrlAndQuery(urljoin(this.baseRoute, domain, '/tags'), query);
```

Comment:
If we have template code like in this method, could we use the Polymorphism principle and delegate boilerplate code that makes HTTP call to a single `static` method that will wait for instance as an argument with particular properties? 
I understand that this implementation will provide a lot of changes but I think about it as an idea for the future changes and it for your choise